generic_server.cc

```diff
@@ -208,7 +208,6 @@ future<> connection::process()
         });
     }).finally([this] {
```

Comment:
> transport: move on_connection_close into connection destructor

patch should explain why, not just what.

Reply:
added explanation